pixeltable/env.py

```diff
@@ -816,6 +818,91 @@ def __init_spacy(self) -> None:
+            self._current_session = None
+
+        if self._current_conn is not None:
+            try:
+                self._current_conn.close()
+            except Exception as e:
+                _logger.error(f'Error closing connection: {e}')
+            self._current_conn = None
+
+        # Stop HTTP server
+        if self._httpd is not None:
+            try:
+                self._httpd.shutdown()
+                self._httpd.server_close()
+            except Exception as e:
+                _logger.error(f'Error stopping HTTP server: {e}')
+            self._httpd = None
+
+        # Stop database server (for local environment)
+        if self._db_server is not None and self._dbms is not None:
+            try:
+                # First terminate all connections to the database
+                if self._db_name is not None:
+                    temp_engine = sql.create_engine(self._dbms.default_system_db_url(), isolation_level='AUTOCOMMIT')
+                    try:
+                        with temp_engine.begin() as conn:
+                            stmt = f"""
+                                SELECT pg_terminate_backend(pg_stat_activity.pid)
+                                FROM pg_stat_activity
+                                WHERE pg_stat_activity.datname = '{self._db_name}'
+                                AND pid <> pg_backend_pid()
+                            """
+                            conn.execute(sql.text(stmt))
+                            _logger.info(f"Terminated all connections to database '{self._db_name}'")
+                    except Exception as e:
+                        _logger.warning(f'Error terminating database connections: {e}')
+                    finally:
+                        temp_engine.dispose()
+            except Exception as e:
+                _logger.error(f'Error stopping database server: {e}')
+            self._db_server = None
+
+        # Dispose of SQLAlchemy engine (after stopping db server)
+        if self._sa_engine is not None:
+            try:
+                self._sa_engine.dispose()
+            except Exception as e:
+                _logger.error(f'Error disposing engine: {e}')
+            self._sa_engine = None
+
+        # Close event loop
+        if self._event_loop is not None:
+            try:
+                if self._event_loop.is_running():
+                    self._event_loop.stop()
+                self._event_loop.close()
+            except Exception as e:
+                _logger.error(f'Error closing event loop: {e}')
+            self._event_loop = None
+
+        # Remove logging handlers
+        for handler in self._logger.handlers[:]:
+            try:
+                handler.close()
+                self._logger.removeHandler(handler)
+            except Exception as e:
+                _logger.error(f'Error removing handler: {e}')
+
+        # Clear temporary directory
+        try:
+            self.clear_tmp_dir()
```

Comment:
Are we certain this is the right thing to do? What if there are other concurrent processes running that are using the same tmp dir? What if there are media files that need to be replayed via a multi phase commit journal?

Reply:
we do it in _setup_up anyway. I'll remove it from cleanup (for data sharing it shouldn't be an issue)
We have to think about sharing directories across multiple worker processes, may be shared directories shouldn't be in tmp dir?